docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -213,32 +329,38 @@ The result will look like this:
 Hello, world
 ```
 
-:::tip
-For consistency between Python and the API, choose to always set optionals
-with a default value, even if it's:
-
-```python
-name: str | None = None
-```
-The above is in fact necessary if the default value needs to be set from the
-function body instead of the signature (dynamic, complex or a mutable value).
-:::
-
 ## Return values
 
 Not only can Dagger Functions use Dagger's core types in their arguments; they can use them in their return value as well.
 
-This opens powerful applications to Dagger Functions. For example, a Dagger Function that builds binaries could take a directory as argument (the source code) and return another directory (containing binaries) or a container image (with the binaries included).
+This opens powerful applications to Dagger Functions. For example, a Dagger Function that builds binaries could take a directory with the source code as argument and return another directory containing just binaries or a container image with the binaries included.
 
 Here's an example of a Dagger Function that accepts a `Directory` containing a Go application's source code as input, compiles it into a binary, and returns a `Container` with the binary:
 
-```python file=./snippets/functions/return-values-container/main.py
+<Tabs groupId="language">
+
+<TabItem value="Go">
+```go file=./snippets/functions/return-values-container/go/main.go
+```
+
+</TabItem>
+
+<TabItem value="Python">
+```python file=./snippets/functions/return-values-container/python/main.py
 ```
+</TabItem>
+
+<TabItem value="TypeScript">
+```typescript file=./snippets/functions/return-values-container/typescript/index.ts
+```
+</TabItem>
+
+</Tabs>
 
 Here is an example call for this Dagger Function:
 
 ```shell
-dagger call build --source=https://github.com/golang/example/#master:hello --os=linux --arch=amd64 terminal
+dagger call build --source=https://github.com/golang/example/#master:hello --os=linux --architecture=amd64 terminal
```

Comment:
We've already used `src` earlier in this page. Maybe use the same name for this argument? Or use `source` in the others as well.

Reply:
updated to use src